### PR TITLE
Integrating Swift Argument Parser

### PR DESCRIPTION
**Pull request by chiragramani**
**Context:** 
The PR aims to integrate [Swift Argument Parser](https://github.com/apple/swift-argument-parser) with Piranha Swift. Swift Argument Parser is now a standard way to accept input from command line and it abstracts all the code to provide the documentation for input commands, custom naming flags, out of box support for flags/arguments/options with transformation that really helps to provide a robust command line input taking away all the pain of manual handling.

Even other Uber swift repos such as CodeSan makes use of the same. This will also help us to have have parity with js piranha and will also help us to easily add new commands or modify existing commands as Piranha evolves.

1. The PR also addresses other minor changes. Keeping the naming of swift directories very much similar to what default SPM packages do.
2. A new target dependency PiranhaKit because test bundle cannot import an executable target and therefore to satisfy the testing needs, this new module was needed.
3. Added unit-tests for the newly added functionality as well as updated the test.sh
4. Also updated the README accordingly. 

Please note: This PR doesn't aim to provide any change in the existing functionality related to AST rewriting.

### CODE REVIEW

File: swift/Sources/PiranhaKit/InputCommand/PiranhaCommand.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                                                           defaultSubcommand: CleanupStaleFlagsCommand.self)
+}
+
+struct CleanupStaleFlagsCommand: ParsableCommand {
+    
+    static var _commandName: String = "cleanup-stale-flags"
+    
+    @Option(name: [.customShort("s"),
+                   .long],
+            help: "Input Source File that needs to run piranha",
```
Comment (mkr-plse) [inline swift/Sources/PiranhaKit/InputCommand/PiranhaCommand.swift:35]: Input source file for Piranha?

**Review by mkr-plse — APPROVED** [review]
Looks good. Minor nit on the help message. Thanks for taking care of this.

File: swift/Sources/PiranhaKit/InputCommand/PiranhaCommand.swift (hunk 1 above)
Comment (chiragramani) [reply swift/Sources/PiranhaKit/InputCommand/PiranhaCommand.swift:35]: Yeah, that does seem better to me too. Thanks for suggesting this - will make the changes!

Comment (chiragramani) [reply swift/Sources/PiranhaKit/InputCommand/PiranhaCommand.swift]: Updated!

**Review by mkr-plse — APPROVED** [review]

**Pull request merged**